### PR TITLE
[refactor] #3622: Granular clippy lints

### PR DESCRIPTION
**Pull request by appetrosyan**
## Description

Drive-by refactoring and experiment with granular linting on a per-crate basis. 

### Linked issue

Partially addresses #3622 

### Checklist

- [ ] Ci checks pass
- [ ] Approval of draft ideas
- [ ] Final execution
- [ ] Final review

### CODE REVIEW

File: crypto/src/hash.rs [hunk 1]
```diff
@@ -82,6 +82,27 @@ impl Hash {
...
+    /// entity, so it's possible to use Rust's type system to prevent
+    /// common errors.
+    ///
+    /// However, most hashes are computed from sources in which the
+    /// type is externally imposed, such as computing the hashes of
+    /// the branches of the Merkle tree, the signatures of a block
+    /// and/or computing the hash from a raw stream of bytes.  **This
+    /// function cannot check your code for type validity, so this
+    /// must be enforced by the user**.
+    pub fn typed<T>(self) -> HashOf<T> {
```
Comment (mversic) [inline crypto/src/hash.rs:101]:
```suggestion
    pub fn typed_unchecked<T>(self) -> HashOf<T> {
```

can we at least have suffix `_unchecked`?

Comment (appetrosyan) [reply crypto/src/hash.rs:101]: If you like.

How about `manually_typed`?

Comment (appetrosyan) [reply crypto/src/hash.rs:101]: Actually, this gives me an idea.

Comment (appetrosyan) [reply crypto/src/hash.rs:101]: I'll rename the postfix function to `untyped_unchecked`.

**Pull request closed without merging**